subnet_router.go

```diff
@@ -0,0 +1,98 @@
+package packngo
+
+import (
+	"path"
+)
+
+const subnetRouterBasePath = "/subnet-routers"
+
+type SubnetRouterService interface {
+	List(projectID string, opts *ListOptions) ([]SubnetRouter, *Response, error)
+	Create(projectID string, input *SubnetRouterCreateRequest) (*SubnetRouter, *Response, error)
+	Get(subnetRouterID string, opts *GetOptions) (*SubnetRouter, *Response, error)
+	Delete(subnetRouterID string) (*Response, error)
+}
+
+type SubnetRouter struct {
+	ID             string                `json:"id"`
+	State          string                `json:"state"`
+	Project        *Project              `json:"project,omitempty"`
+	VirtualNetwork *VirtualNetwork       `json:"virtual_network,omitempty"`
+	IPReservation  *IPAddressReservation `json:"ip_reservation,omitempty"`
+	Href           string                `json:"href"`
+	Created        string                `json:"created_at,omitempty"`
+	Updated        string                `json:"updated_at,omitempty"`
+}
+
+type SubnetRouterServiceOp struct {
+	client *Client
+}
+
+func (s *SubnetRouterServiceOp) List(projectID string, opts *ListOptions) (subnetRouters []SubnetRouter, resp *Response, err error) {
+	type subnetRoutersRoot struct {
+		SubnetRouters []SubnetRouter `json:"subnet_routers"`
+		Meta          meta           `json:"meta"`
+	}
+
+	endpointPath := path.Join(projectBasePath, projectID, subnetRouterBasePath)
+	apiPathQuery := opts.WithQuery(endpointPath)
+
+	for {
+		subset := new(subnetRoutersRoot)
+
+		resp, err = s.client.DoRequest("GET", apiPathQuery, nil, subset)
+		if err != nil {
+			return nil, resp, err
+		}
+
+		subnetRouters = append(subnetRouters, subset.SubnetRouters...)
+
+		if apiPathQuery = nextPage(subset.Meta, opts); apiPathQuery != "" {
+			continue
+		}
+		return
+	}
+
+}
+
+type SubnetRouterCreateRequest struct {
+	VirtualNetworkID      string `json:"virtual_network"`
```

Comment:
The ID suffix here makes sense, it makes packngo consistent about using ID in the field name for ID values.
That said, I looked around, and it seems we have been mostly consistent **not doing this**. (`SpotMarketRequest.FacilityIDs` is an exception).

```
git grep ID.*json
git grep -i virtualnetwork.*json
```

I'll see if we can still reconsider the upstream field names to include `_id`.

Reply:
I tried with _id in JSON and it works. Thanks for resolving.